README.md

```diff
@@ -404,33 +404,87 @@ With the CodePush plugin downloaded and linked, and your app asking CodePush whe
 
 2. When an update is available, how to present it to the end user?
 
-The simplest way to do this is to perform the following in your app's root component:
+The simplest way to do this is to "CodePush-ify" your app's root component. To do so, you can choose one of the following three options:
 
-1. Import the JavaScript module for CodePush:
+**Option 1: Use the [ES7 decorator](https://github.com/wycats/javascript-decorators) syntax:**
 
-    ```javascript
-    import codePush from "react-native-code-push";
-    ```
+```javascript
+import codePush, { codePushify } from "react-native-code-push";
 
-2. Call the `sync` method from within the `componentDidMount` lifecycle event, to initiate a background update on each app start:
+let codePushOptions;
 
-    ```javascript
-    codePush.sync();
+@codePushify(codePushOptions)
+class MyApp extends Component {
+}
+```
+
+*NOTE: Decorators are not yet supported in Babel 6.x pending proposal update.* You may need to enable it by doing the following:
+- Install the [`babel-preset-react-native-stage-0` package](https://github.com/skevy/babel-preset-react-native-stage-0)
+    
+    ```
+    npm install babel-preset-react-native-stage-0 --save-dev
+    ```
+    
+- In your `.babelrc` file, include the following:
+    
+    ```
+    {
+        "presets": ["react-native-stage-0/decorator-support"]
+    }
     ```
 
-If an update is available, it will be silently downloaded, and installed the next time the app is restarted (either explicitly by the end user or by the OS), which ensures the least invasive experience for your end users. If an available update is mandatory, then it will be installed immediately, ensuring that the end user gets it as soon as possible.
+**Option 2: Call `codePushify` on your root component and assign the result to itself after the component declaration:**
+    
+```javascript
+import codePush, { codePushify } from "react-native-code-push";
+
+let codePushOptions;
+
+class MyApp extends Component {
+}
+
+MyApp = codePushify(codePushOptions)(MyApp);
+```
 
-If you would like your app to discover updates more quickly, you can also choose to call `sync` every time the app resumes from the background, by adding the following code (or something equivalent) as part of your app's startup behavior (e.g. your root component's `componentDidMount` method). You can call `sync` as frequently as you would like, so when and where you call it just depends on your personal preference.
+**Option 3: Manually call the `sync` method from within the `componentDidMount` lifecycle event, to initiate a background update on each app start:**
```

Comment:
This is probably something we can discuss - we want everyone new to use the decorator going forward right? Instead of muddying the waters by documenting `sync()` here, would it be worth just adding a link to the 'legacy' version of the README at the top of this one, for people who were already using it?

Then we can just treat `sync()` as an advanced API that you can find if you scroll down and look at the API reference, similarly to `checkForUpdate()` etc.